src/script/lua_api/l_object.cpp

```diff
@@ -606,39 +530,34 @@ int ObjectRef::l_set_animation_frame_speed(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
-	ServerActiveObject *co = getobject(ref);
-	if (co == NULL)
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == nullptr)
 		return 0;
 
-	// Do it
 	if (!lua_isnil(L, 2)) {
-		float frame_speed = lua_tonumber(L, 2);
-		co->setAnimationSpeed(frame_speed);
+		float frame_speed = readParam<float>(L, 2);
+		sao->setAnimationSpeed(frame_speed);
 		lua_pushboolean(L, true);
 	} else {
 		lua_pushboolean(L, false);
 	}
 	return 1;
 }
 
-// set_bone_position(self, std::string bone, v3f position, v3f rotation)
+// set_bone_position(self, bone, position, rotation)
 int ObjectRef::l_set_bone_position(lua_State *L)
 {
 	NO_MAP_LOCK_REQUIRED;
 	ObjectRef *ref = checkobject(L, 1);
-	ServerActiveObject *co = getobject(ref);
-	if (co == NULL) return 0;
-	// Do it
-	std::string bone;
-	if (!lua_isnil(L, 2))
-		bone = readParam<std::string>(L, 2);
-	v3f position = v3f(0, 0, 0);
-	if (!lua_isnil(L, 3))
-		position = check_v3f(L, 3);
-	v3f rotation = v3f(0, 0, 0);
-	if (!lua_isnil(L, 4))
-		rotation = check_v3f(L, 4);
-	co->setBonePosition(bone, position, rotation);
+	ServerActiveObject *sao = getobject(ref);
+	if (sao == nullptr)
+		return 0;
+
+	std::string bone = readParam<std::string>(L, 2);
+	v3f position = !lua_isnil(L, 3) ? check_v3f(L, 3) : v3f(0, 0, 0);
+	v3f rotation = !lua_isnil(L, 4) ? check_v3f(L, 4) : v3f(0, 0, 0);
```

Comment:
In this case, why not just `v3f position = read_v3f(L, 3);`?  If it's nil, it will automatically be zeroed.  Same in many other places.

Reply:
Nice catch, done :)